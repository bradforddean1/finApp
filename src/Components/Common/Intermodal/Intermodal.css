.Intermodal {
	position: fixed;
	display: flex;
	top: 0;
	left: 0;
	align-items: center;
	justify-content: center;
	width: 100vw;
	height: 100vh;
	padding: 24px;
}

.Intermodal .background {
	position: absolute;
	height: 100%;
	width: 100%;
	background: #e5e5e5;
	opacity: 0.8;
}

.intermodal-box {
	position: relative;
	width: 300px;
	min-height: 50px;
	padding: 38px;
	background-color: #000000;
	opacity: 1;
	border-radius: 4px;

	font-family: Futura;
	font-style: normal;
	font-weight: 500;
	font-size: 18px;
	line-height: 24px;
	text-align: center;
	color: #ffffff;
}

.Intermodal p {
	margin-bottom: 16px;
}

.intermodal-button {
	padding: 8px 24px;
	min-width: 95px;
	height: 32px;
	background-color: #46d16d;
	border-radius: 2px;
	margin: 4px;

	font-family: Sofia Pro;
	font-weight: normal;
	font-size: 12px;
	line-height: 134.77%;
	color: #000000;
}
