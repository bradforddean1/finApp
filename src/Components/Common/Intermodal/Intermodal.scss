@use "../../../scss/variables" as v;

.Intermodal {
	position: fixed;
	display: flex;
	top: 0;
	left: 0;
	align-items: center;
	justify-content: center;
	width: 100vw;
	height: 100vh;
	padding: 24px;
}

.Intermodal .background {
	position: absolute;
	height: 100%;
	width: 100%;
	background: v.$loading-bg-color;
	opacity: 0.8;
}

.intermodal-box {
	position: relative;
	width: 300px;
	min-height: 50px;
	padding: 38px;
	background-color: #000000;
	opacity: 1;
	border-radius: 4px;

	font-family: "Hind", sans-serif;
	font-style: normal;
	font-weight: 500;
	font-size: 18px;
	line-height: 24px;
	text-align: center;
	color: #ffffff;
}

.Intermodal p {
	margin-bottom: 16px;
}

.intermodal-button {
	padding: 8px 24px;
	min-width: 95px;
	height: 32px;
	margin: 4px;

	background-color: #000000;
	border-radius: 2px;
	border: 2px solid #46d16d;

	font-family: "Poppins";
	font-weight: normal;
	font-size: 12px;
	line-height: 134.77%;
	color: #46d16d;
}

.intermodal-button:focus {
	background-color: #46d16d;
	color: #000;
	outline: none;
	box-shadow: none;
}
