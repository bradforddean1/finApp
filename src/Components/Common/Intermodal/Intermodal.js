import React, { createElement } from "react";
import PropTypes from "prop-types";
import ButtonX from "../ButtonX/ButtonX";
import "./Intermodal.css";

/**
 * Intermodal overlay
 * @component
 */

function Intermodal(props) {
	const { close, show, children, buttons } = props;

	const makeIntermodalContent = () => {
		if (show) {
			return (
				<div className="Intermodal">
					<div className="background"></div>
					<div className="intermodal-box">
						<ButtonX handleClick={close} />
						{children}
						{!!buttons &&
							buttons.map((button, i) => {
								const { tag, children, className, ...otherProps } = button;
								return createElement(
									tag,
									{
										key: i,
										className: ["intermodal-button", className].join(" "),
										...otherProps,
									},
									children
								);
							})}
					</div>
				</div>
			);
		} else {
			return null;
		}
	};

	return <>{makeIntermodalContent()}</>;
}

Intermodal.propTypes = {
	/**
	 * Callback to handle intermodal close button click
	 */
	close: PropTypes.func.isRequired,
	/**
	 * Is intermodal being displayed
	 */
	show: PropTypes.bool,
	/**
	 * List of action Buttons to be implemented
	 */
	buttons: PropTypes.arrayOf(
		PropTypes.shape({
			tag: PropTypes.string.isRequired,
			className: PropTypes.string,
			onClick: PropTypes.func,
			href: PropTypes.string,
			children: PropTypes.oneOfType([
				PropTypes.string,
				PropTypes.element,
				PropTypes.arrayOf(PropTypes.element),
			]),
		})
	),
	children: PropTypes.element,
};

Intermodal.defaultProps = {
	show: false,
};

export default Intermodal;
