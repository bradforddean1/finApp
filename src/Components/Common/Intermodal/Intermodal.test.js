import React from "react";
import ReactDOM from "react-dom";
import Intermodal from "./Intermodal";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("Intermodal Component", () => {
	it("Renders", () => {
		const div = document.createElement("div");
		ReactDOM.render(<Intermodal />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it("Renders the Default State", () => {
		const wrapper = shallow(<Intermodal />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	const props = {
		close: () => {
			return null;
		},
		show: true,
		buttons: [
			{
				tag: "a",
				className: null,
				href: "/",
				children: "Test Button Content",
			},
		],
		children: "Test Content",
	};

	it("Renders given props", () => {
		const wrapper = shallow(<Intermodal {...props} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
