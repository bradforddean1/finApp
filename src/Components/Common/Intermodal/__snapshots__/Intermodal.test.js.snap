// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intermodal Component Renders given props 1`] = `
<Fragment>
  <div
    className="Intermodal"
  >
    <div
      className="background"
    />
    <div
      className="intermodal-box"
    >
      <ButtonX
        handleClick={[Function]}
      />
      Test Content
      <a
        className="intermodal-button "
        href="/"
        key="0"
      >
        Test Button Content
      </a>
    </div>
  </div>
</Fragment>
`;

exports[`Intermodal Component Renders the Default State 1`] = `<Fragment />`;
